core/src/main/java/org/apache/druid/java/util/common/parsers/JSONPathParser.java

```diff
@@ -42,7 +42,11 @@ public class JSONPathParser implements Parser<String, Object>
   public JSONPathParser(JSONPathSpec flattenSpec, ObjectMapper mapper, boolean keepNullColumns)
   {
     this.mapper = mapper == null ? new ObjectMapper() : mapper;
-    this.flattener = ObjectFlatteners.create(flattenSpec, new JSONFlattenerMaker(keepNullColumns));
+    this.flattener = ObjectFlatteners.create(
+        flattenSpec,
+        new JSONFlattenerMaker(keepNullColumns),
+        false
```

Comment:
I'm not certain I know why this is hard-coding to false.  Is that covered in a comment or javadoc or something?

Reply:
this and other `Parser` based implementations are hard coded to false to retain existing behavior for Hadoop ingestion. `Parser` are not created with a `InputRowSchema` the same way as `InputEntityReader` are, so I don't have easy access to a `TuningConfig`, and I wasn't very motivated to fix this up for Hadoop.